File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/BondReference.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class BondReference : ISerialized
    {
        public BondReference(XmlNode xmlNode)
        {
            XmlNode bondNode = xmlNode.SelectSingleNode("bond");
            
            if (bondNode != null)
            {
                if (bondNode.Attributes["href"] != null || bondNode.Attributes["id"] != null) 
                {
                    if (bondNode.Attributes["id"] != null) 
                    {
                        bondIDRef_ = bondNode.Attributes["id"].Value;
                        Bond ob = new Bond(bondNode);
                        IDManager.SetID(bondIDRef_, ob);
                    }
                    else if (bondNode.Attributes["href"] != null)
                    {
                        bondIDRef_ = bondNode.Attributes["href"].Value;
                    }
                    else
                    {
                        bond_ = new Bond(bondNode);
                    }
                }
                else
                {
                    bond_ = new Bond(bondNode);
                }
            }
            
        
            XmlNode conditionPrecedentBondNode = xmlNode.SelectSingleNode("conditionPrecedentBond");
            
            if (conditionPrecedentBondNode != null)
            {
                if (conditionPrecedentBondNode.Attributes["href"] != null || conditionPrecedentBondNode.Attributes["id"] != null) 
                {
                    if (conditionPrecedentBondNode.Attributes["id"] != null) 
                    {
                        conditionPrecedentBondIDRef_ = conditionPrecedentBondNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(conditionPrecedentBondNode);
                        IDManager.SetID(conditionPrecedentBondIDRef_, ob);
                    }
                    else if (conditionPrecedentBondNode.Attributes["href"] != null)
                    {
                        conditionPrecedentBondIDRef_ = conditionPrecedentBondNode.Attributes["href"].Value;
                    }
                    else
                    {
                        conditionPrecedentBond_ = new XsdTypeBoolean(conditionPrecedentBondNode);
                    }
                }
                else
                {
                    conditionPrecedentBond_ = new XsdTypeBoolean(conditionPrecedentBondNode);
                }
            }
            
        
            XmlNode discrepancyClauseNode = xmlNode.SelectSingleNode("discrepancyClause");
            
            if (discrepancyClauseNode != null)
            {
                if (discrepancyClauseNode.Attributes["href"] != null || discrepancyClauseNode.Attributes["id"] != null) 
                {
                    if (discrepancyClauseNode.Attributes["id"] != null) 
                    {
                        discrepancyClauseIDRef_ = discrepancyClauseNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(discrepancyClauseNode);
                        IDManager.SetID(discrepancyClauseIDRef_, ob);
                    }
                    else if (discrepancyClauseNode.Attributes["href"] != null)
                    {
                        discrepancyClauseIDRef_ = discrepancyClauseNode.Attributes["href"].Value;
                    }
                    else
                    {
                        discrepancyClause_ = new XsdTypeBoolean(discrepancyClauseNode);
                    }
                }
                else
                {
                    discrepancyClause_ = new XsdTypeBoolean(discrepancyClauseNode);
                }
            }
            
        
        }
        
    
        #region Bond_
        private Bond bond_;
        public Bond Bond_
        {
            get
            {
                if (this.bond_ != null)
                {
                    return this.bond_; 
                }
                else if (this.bondIDRef_ != null)
                {
                    bond_ = IDManager.getID(bondIDRef_) as Bond;
                    return this.bond_; 
                }
                else
                {
                      return this.bond_; 
                }
            }
            set
            {
                if (this.bond_ != value)
                {
                    this.bond_ = value;
                }
            }
        }
        #endregion
        
        public string bondIDRef_ { get; set; }
        #region ConditionPrecedentBond_
        private XsdTypeBoolean conditionPrecedentBond_;
        public XsdTypeBoolean ConditionPrecedentBond_
        {
            get
            {
                if (this.conditionPrecedentBond_ != null)
                {
                    return this.conditionPrecedentBond_; 
                }
                else if (this.conditionPrecedentBondIDRef_ != null)
                {
                    conditionPrecedentBond_ = IDManager.getID(conditionPrecedentBondIDRef_) as XsdTypeBoolean;
                    return this.conditionPrecedentBond_; 
                }
                else
                {
                      return this.conditionPrecedentBond_; 
                }
            }
            set
            {
                if (this.conditionPrecedentBond_ != value)
                {
                    this.conditionPrecedentBond_ = value;
                }
            }
        }
        #endregion
        
        public string conditionPrecedentBondIDRef_ { get; set; }
        #region DiscrepancyClause_
        private XsdTypeBoolean discrepancyClause_;
        public XsdTypeBoolean DiscrepancyClause_
        {
            get
            {
                if (this.discrepancyClause_ != null)
                {
                    return this.discrepancyClause_; 
                }
                else if (this.discrepancyClauseIDRef_ != null)
                {
                    discrepancyClause_ = IDManager.getID(discrepancyClauseIDRef_) as XsdTypeBoolean;
                    return this.discrepancyClause_; 
                }
                else
                {
                      return this.discrepancyClause_; 
                }
            }
            set
            {
                if (this.discrepancyClause_ != value)
                {
                    this.discrepancyClause_ = value;
                }
            }
        }
        #endregion
        
        public string discrepancyClauseIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/BondReference.cpp
// BondReference.cpp 
#include "BondReference.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

BondReference::BondReference(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //bondNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* bondNode = xmlNode->FirstChildElement("bond");

   if(bondNode){bondIsNull_ = false;}
   else{bondIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- bondNode , address : " << bondNode << std::endl;
   #endif
   if(bondNode)
   {
      if(bondNode->Attribute("href") || bondNode->Attribute("id"))
      {
          if(bondNode->Attribute("id"))
          {
             bondIDRef_ = bondNode->Attribute("id");
             bond_ = boost::shared_ptr<Bond>(new Bond(bondNode));
             bond_->setID(bondIDRef_);
             IDManager::instance().setID(bondIDRef_,bond_);
          }
          else if(bondNode->Attribute("href")) { bondIDRef_ = bondNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { bond_ = boost::shared_ptr<Bond>(new Bond(bondNode));}
   }

   //conditionPrecedentBondNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* conditionPrecedentBondNode = xmlNode->FirstChildElement("conditionPrecedentBond");

   if(conditionPrecedentBondNode){conditionPrecedentBondIsNull_ = false;}
   else{conditionPrecedentBondIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- conditionPrecedentBondNode , address : " << conditionPrecedentBondNode << std::endl;
   #endif
   if(conditionPrecedentBondNode)
   {
      if(conditionPrecedentBondNode->Attribute("href") || conditionPrecedentBondNode->Attribute("id"))
      {
          if(conditionPrecedentBondNode->Attribute("id"))
          {
             conditionPrecedentBondIDRef_ = conditionPrecedentBondNode->Attribute("id");
             conditionPrecedentBond_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(conditionPrecedentBondNode));
             conditionPrecedentBond_->setID(conditionPrecedentBondIDRef_);
             IDManager::instance().setID(conditionPrecedentBondIDRef_,conditionPrecedentBond_);
          }
          else if(conditionPrecedentBondNode->Attribute("href")) { conditionPrecedentBondIDRef_ = conditionPrecedentBondNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { conditionPrecedentBond_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(conditionPrecedentBondNode));}
   }

   //discrepancyClauseNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* discrepancyClauseNode = xmlNode->FirstChildElement("discrepancyClause");

   if(discrepancyClauseNode){discrepancyClauseIsNull_ = false;}
   else{discrepancyClauseIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- discrepancyClauseNode , address : " << discrepancyClauseNode << std::endl;
   #endif
   if(discrepancyClauseNode)
   {
      if(discrepancyClauseNode->Attribute("href") || discrepancyClauseNode->Attribute("id"))
      {
          if(discrepancyClauseNode->Attribute("id"))
          {
             discrepancyClauseIDRef_ = discrepancyClauseNode->Attribute("id");
             discrepancyClause_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(discrepancyClauseNode));
             discrepancyClause_->setID(discrepancyClauseIDRef_);
             IDManager::instance().setID(discrepancyClauseIDRef_,discrepancyClause_);
          }
          else if(discrepancyClauseNode->Attribute("href")) { discrepancyClauseIDRef_ = discrepancyClauseNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { discrepancyClause_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(discrepancyClauseNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Bond> BondReference::getBond()
{
   if(!this->bondIsNull_){
        if(bondIDRef_ != ""){
             return boost::shared_static_cast<Bond>(IDManager::instance().getID(bondIDRef_));
        }else{
             return this->bond_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Bond>();
   }
}
boost::shared_ptr<XsdTypeBoolean> BondReference::getConditionPrecedentBond()
{
   if(!this->conditionPrecedentBondIsNull_){
        if(conditionPrecedentBondIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(conditionPrecedentBondIDRef_));
        }else{
             return this->conditionPrecedentBond_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> BondReference::getDiscrepancyClause()
{
   if(!this->discrepancyClauseIsNull_){
        if(discrepancyClauseIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(discrepancyClauseIDRef_));
        }else{
             return this->discrepancyClause_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

